server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DLPConfigurationRoutesTest.java

```diff
@@ -0,0 +1,571 @@
+                .delete(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.NO_CONTENT_204);
+
+            String retrievedBody = with()
+                .get(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body().asString();
+
+            assertThatJson(retrievedBody).isEqualTo("{\"rules\":[]}");
+        }
+
+        @Test
+        void deleteShouldRemoveOnlyConfigurationsFromCorrespondingDomain() {
+            dlpStore.store(SENDER_DOMAIN, CONFIGURATION_ITEMS);
+            dlpStore.store(SENDER_DOMAIN_2, CONFIGURATION_ITEMS_FOR_DOMAIN_2);
+
+            when()
+                .delete(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.NO_CONTENT_204);
+
+            String retrievedBody = with()
+                .get(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body().asString();
+
+            assertThatJson(retrievedBody).isEqualTo("{\"rules\":[]}");
+
+            String retrievedBodyDomain2 = when()
+                .get(DOMAIN_2)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body().asString();
+
+            assertThatJson(retrievedBodyDomain2)
+                .isEqualTo(
+                    "{\"rules\": [" +
+                    "    {" +
+                    "        \"id\": \"3\"," +
+                    "        \"expression\": \"apache.org\"," +
+                    "        \"explanation\": null," +
+                    "        \"targetsSender\": true," +
+                    "        \"targetsRecipients\": false," +
+                    "        \"targetsContent\": false" +
+                    "    }" +
+                    "]}");
+        }
+
+        @Test
+        void deleteShouldReturnNotFoundWhenDomainNotInList() {
+            when()
+                .delete("strange.com")
+            .then()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(HttpStatus.NOT_FOUND_404))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("'strange.com' is not managed by this James server"));
+        }
+
+        @Test
+        void deleteShouldReturnBadRequestWhenDomainIsNotValid() {
+            when()
+                .delete("dr@strange.com")
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(HttpStatus.BAD_REQUEST_400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("Invalid request for domain: dr@strange.com"));
```

Comment:
cause should also be asserted so that I'm sure we explain WHY the request is not valid.

Reply:
idem, `IlleagalAgrumentException` with a null message

Reply:
Idem, that's not a OK behaviour.